src/wp-includes/option.php

```diff
@@ -1106,6 +1113,71 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ * Determine the appropriate autoload value based on input.
+ *
+ * This function checks the provided autoload value and returns a standardized
+ * value ('yes', 'no', 'default-yes', or 'default-no') based on specific conditions.
+ *
+ * @since 6.4.0
+ *
+ * @param mixed $autoload The autoload value to evaluate.
+ *
+ * @return string The standardized autoload value: 'yes', 'no', 'default-yes', or 'default-no'.
+ */
+function sanitize_autoload( $autoload ) {
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'yes' : 'no';
+	}
+
+	switch ( $autoload ) {
+		case 'no':
+		case 'yes':
+		case 'default-yes':
+		case 'default-no':
+			return $autoload;
+		default:
+			return 'default-yes';
+	}
+}
+
+/**
+ * Check the size of an option value and compare it to the maximum allowed size.
+ *
+ * This function serializes the given value and checks its size against the maximum allowed
+ * option size defined by the 'max_option_size' filter.
+ *
+ * @since 6.4.0
```

Comment:
As you've suggested in https://github.com/WordPress/wordpress-develop/pull/4881#discussion_r1328880918, let's mark this as private.
```suggestion
 * @since 6.4.0
 * @access private
```